dvc/tree/__init__.py

```diff
@@ -78,5 +78,11 @@ def _resolve_remote_refs(repo, remote_conf):
 
 
 def get_cloud_tree(repo, **kwargs):
+    from dvc.config import SCHEMA, ConfigError, Invalid
+
     remote_conf = _get_conf(repo, **kwargs)
+    try:
+        remote_conf = SCHEMA["remote"][str](remote_conf)
```

Comment:
One might ask on why not to do this automatically during config verification (e.g. dvc/config.py::ByUrl), but that one has a very limited scope, so it is not trivial to make it resolve urls like this. Plus that would be static, while this resolution is dynamic.